AdobeStockAsset/Test/Integration/Model/SaveAssetTest.php

```diff
@@ -0,0 +1,167 @@
+
+/**
+ * Provides integration tests for saving and Adobe Stock asset.
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class SaveAssetTest extends TestCase
+{
+    private const MEDIA_GALLERY_ASSET_PATH = 'some/path.jpg';
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    private $categoryRepository;
+
+    /**
+     * @var CreatorRepositoryInterface
+     */
+    private $creatorRepository;
+
+    /**
+     * @var AssetRepositoryInterface
+     */
+    private $assetRepository;
+
+    /**
+     * @var DeleteByPathInterface
+     */
+    private $deleteMediaAssetByPath;
+
+    /**
+     * @var SaveAssetInterface
+     */
+    private $saveAsset;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp()
+    {
+        $this->categoryRepository = Bootstrap::getObjectManager()->get(CategoryRepositoryInterface::class);
+        $this->creatorRepository = Bootstrap::getObjectManager()->get(CreatorRepositoryInterface::class);
+        $this->assetRepository = Bootstrap::getObjectManager()->get(AssetRepositoryInterface::class);
+        $this->deleteMediaAssetByPath = Bootstrap::getObjectManager()->get(DeleteByPathInterface::class);
+        $this->saveAsset = Bootstrap::getObjectManager()->get(SaveAssetInterface::class);
+    }
+
+    /**
+     * Test save an Adobe Stock asset.
+     */
+    public function testExecute(): void
+    {
+        /** @var AssetInterface $asset */
+        $asset = $this->prepareAsset();
+        $this->saveAsset->execute($asset);
+        $expectedAsset = $this->assetRepository->getById($asset->getId());
+
+        $this->assertEquals($expectedAsset->getId(), $asset->getId());
+        $this->assertEquals($expectedAsset->getCategoryId(), $asset->getCategory()->getId());
+        $this->assertEquals($expectedAsset->getCreatorId(), $asset->getCreator()->getId());
+        $this->assertEquals($expectedAsset->getMediaGalleryId(), $asset->getMediaGalleryId());
+
+        $this->cleanUpEntries($expectedAsset);
+    }
+
+    /**
+     * Prepare an Adobe Stock asset test object.
+     */
+    public function prepareAsset(): AssetInterface
+    {
+        /** @var MediaAssetFactory $mediaAssetFactory */
+        $mediaAssetFactory = Bootstrap::getObjectManager()->get(MediaAssetFactory::class);
+        /** @var MediaAsset $mediaAsset */
+        $mediaAsset = $mediaAssetFactory->create(
+            [
+                'data' => [
+                    'path' => self::MEDIA_GALLERY_ASSET_PATH,
+                    'title' => 'Web API test image',
+                    'source' => 'Adobe Stock',
+                    'content_type' => 'image/jpeg',
+                    'width' => 6529,
+                    'height' => 4355,
+                    'size' => 424242
+                ]
+            ]
+        );
+        /** @var SaveInterface $mediaSave */
+        $mediaSave = Bootstrap::getObjectManager()->get(SaveInterface::class);
+        $mediaId = $mediaSave->execute($mediaAsset);
```

Comment:
Please use fixture for media asset creation (including rollback).

Reply:
I use fixtures for all other cases when we need to have an entity in the DB. In that one, I do not think that using a fixture is a benefit. We need to test that the asset object will be set and creating it during the test completely reflects the thing we need to test - asset creation process. Otherwise who the fixture can help in that case?

Reply:
My idea is similar to yours: https://github.com/magento/magento2/pull/27499/files#diff-de25d0fa5b838d2dfe3adc3c8d033e81 Where you made an end-to-en test cover the save process from the needed object initialization and save. Instead of rollback use repository to delete item. I think this is a good example of an integration test.

But if it is not and I will, of course, implement a rollback and other stuff but I would really appreciate if you will explain how asset fixture can help in that case and why in the mentioned test you didn't use it. @sivaschenko

Reply:
This particular test is for AdobeStockAsset, so testing the media asset (from a different module) is not necessary.
Media asset can be created by fixture and deleted by fixture rollback. It should be loaded in the test and than and used for Adobe Stock asset creation